### PR TITLE
Update jekyll and jekyll-assets

### PR DESCRIPTION
**Pull request by hursey013**
Related to https://github.com/18F/federalist-uswds-jekyll/issues/186, maybe https://github.com/18F/federalist-uswds-jekyll/issues/223?

**What this PR does:**
- Updates Jekyll (and other dependencies) to latest versions
- Updates `jekyll-assets` to use GitHub repository which contains compatibility fixes for Jekyll > 4.0
- Removes USWDS Jekyll plugin which copied static assets since it is now handled by asset pipeline
- Exposes USWDS theme sass files for easier configuration
- Overrides select USWDS sass files to support asset pipeline for USWDS assets
- Updates various includes and layouts to use correct asset pipeline path
- Use the `--livereload` flag w/ `npm start` for better development experience

While this does resolve the underlying issues, it's still dependent upon `jekyll-assets` which does not appear to be actively maintained. This PR is a simple drop in update that will extend the life of the Jekyll starter, however future PRs should likely explore more modern approaches to compiling assets.

Also, as a bonus, using the asset pipeline for USWDS reduces the number of files that need to be uploaded (~2000 files down to 130), although not a huge impact on the total build/upload time.

### CODE REVIEW

**Review by davemcorwin — COMMENTED** [review]
@hursey013 well, this is awesome!
- being able to only pull what we use from USWDS is a big win, and now all those assets get fingerprinted as well!
- breaking out the theming stuff makes it easier to see and change, but harder to track what IS changed from the default, not sure how big of a concern this is
- with the "overrides", is there a risk of USWDS changing implementation details that would break them?
- now that you have really dug into this, is there some parameterization that USWDS could enable so that we wouldn't have to override these functions?

**Review by davemcorwin — COMMENTED** [review]
also, looks like there is some auto-formatting going on. I don't have an opinion on the actual formatting, but we should try to be careful about including those changes since it adds some noise to the PR review. In VSCode, you can do a cmd-k cmd-s to save without formatting.

**Comment by hursey013** [comment]
> * breaking out the theming stuff makes it easier to see and change, but harder to track what IS changed from the default, not sure how big of a concern this is
The theme files are a direct copy/paste from the latest USWDS, with the exception of two variables, the font and img paths. I could set those variables elsewhere though, that way if someone copies a fresh set of theme files it won't break anything.
> * with the "overrides", is there a risk of USWDS changing implementation details that would break them?
There is a risk, but it should be minimal. We could pin the USWDS version to be safe, and only upgrade once we've verified everything works as expected, but that does add some extra overhead.
> * now that you have really dug into this, is there some parameterization that USWDS could enable so that we wouldn't have to override these functions?
It looks like the thought is [already there](https://github.com/uswds/uswds/blob/1908d1391bc59410624ca1934cc70b7404e8f443/src/stylesheets/core/mixins/_font-face.scss#L5), but basically we'd need the ability to set a Sass variable to inform USWDS that we'd like to use the asset-pipeline syntax rather than importing files directly into the CSS with `url()`. Doing this would eliminate the need for those overrides.

File: Gemfile [hunk 1]
```diff
@@ -1,20 +1,17 @@
-ruby '>= 2.7.4'
-
 source "https://rubygems.org"
+ruby '>= 2.7.3'
```
Comment (davemcorwin) [inline Gemfile:2]: Was downgrading to 2.7.3 a requirement?

**Comment by davemcorwin** [comment]
> > * now that you have really dug into this, is there some parameterization that USWDS could enable so that we wouldn't have to override these functions?
> >   It looks like the thought is [already there](https://github.com/uswds/uswds/blob/1908d1391bc59410624ca1934cc70b7404e8f443/src/stylesheets/core/mixins/_font-face.scss#L5), but basically we'd need the ability to set a Sass variable to inform USWDS that we'd like to use the asset-pipeline syntax rather than importing files directly into the CSS with `url()`. Doing this would eliminate the need for those overrides.

I don't know enough about scss, but is it generic enough that USWDS could use some function `custom-url()` where the default implementation is `url()` but where it could be overridden by any custom function?

File: Gemfile (hunk 1 above)
Comment (hursey013) [reply Gemfile:2]: No, meant to change this back - for some reason my local env w/ chruby and ruby-install did not want to use 2.7.4.

**Comment by hursey013** [comment]
> I don't know enough about scss, but is it generic enough that USWDS could use some function `custom-url()` where the default implementation is `url()` but where it could be overridden by any custom function?

Yeah that's my understanding - something along the lines of this I believe https://css-tricks.com/snippets/sass/simple-asset-helper-functions/

**Review by davemcorwin — APPROVED** [review]
LGTM

**Pull request merged**